### PR TITLE
[Spotless] Applying Google Code Format for core/src/main files #4

### PR DESCRIPTION
**Pull request by MitchellGale**
### Description

This PR applies Spotless on Java code for:
'core/src/main/java/org/opensearch/sql/ast/*.java',
'core/src/main/java/org/opensearch/sql/DataSourceSchemaName.java',
'core/src/test/java/org/opensearch/sql/ast/*.java',
'core/src/test/java/org/opensearch/sql/data/*.java',
'core/src/test/java/org/opensearch/sql/config/*.java',
'core/src/test/java/org/opensearch/sql/analysis/*.java'

Disables core checkstyle and enables spotless check for files that are fixed in this PR.
 
### Issues Resolved
[[List any issues this PR will resolve]](https://github.com/opensearch-project/sql/issues/1101)
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -95,7 +100,7 @@ spotless {
 //        removeUnusedImports()
 //        trimTrailingWhitespace()
 //        endWithNewline()
-//        googleJavaFormat('1.17.0').reflowLongStrings().groupArtifact('com.google.googlejavaformat:google-java-format')
```
Comment (acarbonetto) [inline build.gradle:98]: can probably delete this now?

File: core/build.gradle [hunk 2]
```diff
@@ -34,6 +34,9 @@ repositories {
     mavenCentral()
 }
 
+checkstyleTest.ignoreFailures = true
```
Comment (acarbonetto) [inline core/build.gradle:37]: maybe add a TODO with the issue #?

File: core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java [hunk 3]
```diff
@@ -249,27 +247,19 @@ public static Function function(String funcName, UnresolvedExpression... funcArg
   }
 
   /**
-   * CASE
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java:252]: can we ignore spotless here?

File: core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java [hunk 4]
```diff
@@ -249,27 +247,19 @@ public static Function function(String funcName, UnresolvedExpression... funcArg
...
     return caseWhen(null, elseClause, whenClauses);
   }
 
   /**
-   * CASE case_value_expr
-   *     WHEN compare_expr THEN result_expr
-   *     [WHEN compare_expr THEN result_expr] ...
-   *     [ELSE result_expr]
-   * END
+   * CASE case_value_expr WHEN compare_expr THEN result_expr [WHEN compare_expr THEN result_expr]
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java:258]: can we ignore spotless here?

File: core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java [hunk 5]
```diff
@@ -81,17 +75,12 @@ public Optional<String> first() {
...
-   * {@code
-   *   QualifiedName name = ...
-   *   Optional<String> first = name.first();
-   *   if (first.isPresent()) {
-   *     name.rest() ...
-   *   }
-   * }
-   * @return  rest part(s)
+   * Get rest parts of the qualified name. Assume that there must be remaining parts so caller is
+   * responsible for the check (first() or size() must be called first). For example: {@code
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java:79]: can we ignore spotless here?

**Review by acarbonetto — CHANGES_REQUESTED** [review]
@MitchellGale looks like there are a couple of commits form vamsi in your PR.  Can you rebase and remove them please?

File: build.gradle (hunk 1 above)
Comment (MitchellGale) [reply build.gradle:98]: Delete which part?

File: core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java (hunk 4 above)
Comment (MitchellGale) [reply core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java]: Fixed

File: core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java (hunk 5 above)
Comment (MitchellGale) [reply core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java]: Fixed

File: core/build.gradle (hunk 2 above)
Comment (MitchellGale) [reply core/build.gradle:39]: Done!

File: core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java [hunk 6]
```diff
@@ -249,10 +249,10 @@ public static Function function(String funcName, UnresolvedExpression... funcArg
   }
 
   /**
-   * CASE
-   *     WHEN search_condition THEN result_expr
-   *     [WHEN search_condition THEN result_expr] ...
-   *     [ELSE result_expr]
+   * CASE<br>
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java:252]: This file already included in #312 (part 1)

File: core/src/main/java/org/opensearch/sql/ast/tree/TableFunction.java [hunk 7]
```diff
@@ -3,33 +3,27 @@
...
 import lombok.ToString;
 import org.opensearch.sql.ast.AbstractNodeVisitor;
-import org.opensearch.sql.ast.expression.Let;
 import org.opensearch.sql.ast.expression.QualifiedName;
 import org.opensearch.sql.ast.expression.UnresolvedExpression;
 
-/**
- * ASTNode for Table Function.
- */
+/** ASTNode for Table Function. */
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/ast/tree/TableFunction.java:18]:
```suggestion
/** AST Node for Table Function. */
```

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

File: core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java (hunk 6 above)
Comment (MitchellGale) [reply core/src/main/java/org/opensearch/sql/ast/dsl/AstDSL.java]: Great catch! Entire `**/ast` directory was included in both. I removed it from this repo and will use it from PR #1.

File: core/src/main/java/org/opensearch/sql/ast/tree/TableFunction.java (hunk 7 above)
Comment (MitchellGale) [reply core/src/main/java/org/opensearch/sql/ast/tree/TableFunction.java]: Addressed in #312

**Pull request merged**